stdlib/LinearAlgebra/src/matmul.jl

```diff
@@ -6,8 +6,18 @@ matprod(x, y) = x*y + x*y
 
 # Dot products
 
-vecdot(x::Union{DenseArray{T},StridedVector{T}}, y::Union{DenseArray{T},StridedVector{T}}) where {T<:BlasReal} = BLAS.dot(x, y)
-vecdot(x::Union{DenseArray{T},StridedVector{T}}, y::Union{DenseArray{T},StridedVector{T}}) where {T<:BlasComplex} = BLAS.dotc(x, y)
+_vecdot(x::AbstractMatrix{T}, y::AbstractMatrix{T}, ::DenseColumnMajor, ::DenseColumnMajor) where {T<:BlasReal} = BLAS.dot(x, y)
```

Comment:
This previously supported `vecdot(rand(3,3,3), rand(27))` and `vecdot(rand(3,3,3), rand(3,3,3))`.  It doesn't look like it will anymore.  I guess it's not tested?

Reply:
I guess not. I agree that `vecdot(rand(3,3,3), rand(27))` should work.